pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -361,6 +374,38 @@ func (r *ReconcileClusterDeployment) updateClusterDeploymentStatus(cd *hivev1.Cl
 		cd.Status.WebConsoleURL = u.String()
 	}
 
+	// Update remote cluster's version into our status
+	if cd.Status.AdminKubeconfigSecret.Name != "" {
+		kubeconfig, err := r.loadSecretData(cd.Status.AdminKubeconfigSecret.Name, cd.Namespace, adminKubeconfigKey)
+		if err != nil {
+			cdLog.WithError(err).Error("error loading remote cluster's kubeconfig")
+			return err
+		}
+
+		remoteClusterAPIClient, err := r.remoteClusterAPIClientBuilder(kubeconfig)
+		if err != nil {
+			cdLog.WithError(err).Error("error building remote cluster-api client connection")
+			return err
+		}
+
+		remoteClusterVersion := &openshiftapiv1.ClusterVersion{}
+		err = remoteClusterAPIClient.Get(context.Background(),
+			types.NamespacedName{Name: clusterVersionObjectName},
+			remoteClusterVersion)
+		if err != nil {
+			cdLog.WithError(err).Error("error fetching remote clusterversion object")
+			return err
+		}
+
+		cdLog.Debugf("remote desired version for cluster %v: %+v", cd.Name, remoteClusterVersion.Status.Current)
```

Comment:
If you want to report the desired version, it should be remoteCusterVersion.Spec.DesiredUpdate.Version

Reply:
yep. i meant to print the current version out of status, not the desired version in spec